bonus.md

```diff
@@ -3,11 +3,18 @@
 Edit this file to answer the following questions. Give examples. You may need to do a bit of research. :)
 
 1. Why is Hash.fetch useful?
+Using #fetch helps "ensure that the given key exists in a hash (or array). This can eliminate confusing NoMethodErrors later in the code"
+
+"You can use the optional block argument to #fetch to either return an alternate value, or to take some arbitrary action when a value is missing"
 
 2. What is the disadvantage of single inheritance, and what's a possible solution to that problem?
 
+"The answer is that inheritance violates encapsulation.  In other words, by inheriting from a class you're coupling yourself with any number of potential implementation details of the parent; all of which are completely avoided by using composition instead."
+
```

Comment:
Interesting response!  I think the question is asking about the disadvantages of single inheritance vs multiple inheritance, but since that isn't something we've really talked about it's totally reasonable to answer it as a question about disadvantages of inheritance vs composition.